pkgs/top-level/java-packages.nix

```diff
@@ -45,7 +45,11 @@ in {
     mkBootstrap = adoptopenjdk: path: args:
       /* adoptopenjdk not available for i686, so fall back to our old builds for bootstrapping */
       if   adoptopenjdk.jdk-hotspot.meta.available
-      then adoptopenjdk.jdk-hotspot.override { gtkSupport = false; }
+      then
+        # only linux has the gtkSupport option
+        if stdenv.isLinux
+        then adoptopenjdk.jdk-hotspot.override { gtkSupport = false; }
```

Comment:
Ideally adoptopenjdk.jdk-hotspot should take the flag on all platforms and complain if gtkSupport is true whne on darwin.

Reply:
Or could it take `headless ? false`?
While adoptopenjdk does not implement a headless optimization on darwin, a missing disk space optimization is harmless to ignore, so that would remove the need for a warning (unlike enabling a non-existing component, which I agree would need a warning).